src/palace/manager/api/admin/controller/library_settings.py

```diff
@@ -183,6 +183,178 @@ def create_library(self, short_name: str) -> tuple[Library, bool]:
         )
         return library, is_new
 
+    def import_libraries(self) -> Response | ProblemDetail:
+        """Import multiple libraries from JSON data. Libraries that already exist will not be updated.
+
+        Expected JSON format:
+        {
+            "libraries": [
+                {
+                    "name": "Library Name",
+                    "short_name": "lib_short",
+                    "website_url": "https://example.com",
+                    "patron_support_email": "support@example.com"
+                },
+                ...
+            ]
+        }
+        """
+        # Only system admins can create new libraries
+        self.require_system_admin()
+
+        try:
+            data = flask.request.json
+            if not data or "libraries" not in data:
+                raise ProblemDetailException(
+                    problem_detail=INCOMPLETE_CONFIGURATION.detailed(
+                        "Request must include 'libraries' field."
+                    )
+                )
+
+            libraries_data = data["libraries"]
+            if not isinstance(libraries_data, list):
+                raise ProblemDetailException(
+                    problem_detail=INCOMPLETE_CONFIGURATION.detailed(
+                        "'libraries' must be a list."
+                    )
+                )
+
+            created = []
+            skipped = []
+            errors = []
+
+            for idx, library_data in enumerate(libraries_data):
+                try:
+                    result = self._import_single_library(library_data)
+                    if result["is_new"]:
+                        created.append(result)
+                    else:
+                        skipped.append(result)
+                except ProblemDetailException as e:
+                    errors.append(
+                        {
+                            "index": idx,
+                            "data": library_data,
+                            "error": str(e.problem_detail.detail),
+                        }
+                    )
+                except Exception as e:
```

Comment:
This `except Exception as e:` is too broad and should be avoided. Here's why:

  1. **Hides unexpected errors**: Catches programming errors (`AttributeError`, `TypeError`, etc.) that should fail fast rather than being silently added to the errors list. This makes bugs harder to debug.
  2. **Loses context**: `str(e)` often doesn't provide enough information to diagnose issues. Stack traces and exception types are lost.
  3. **Treats all errors the same**: Validation errors, database errors, and programming bugs all get the same treatment, but they likely require different handling.
  4. **Makes the code unpredictable**: Future code changes might raise new exception types that shouldn't be caught here but will be anyway.